django_guid/middleware.py

```diff
@@ -12,7 +12,10 @@
 
 class GuidMiddleware(object):
     """
-    Injects a GUID into the thread that is accessible from anywhere in the Django app
+    Checks for an existing GUID (correlation ID) in a request's headers.
+    If a header value is found, the value is validated as a GUID and stored, before the request is passed to the next middleware.
+    If no value is found, or one is found but is invalid, we generate and store a new GUID.
```

Comment:
Maybe also add that we're storing the GUID in the thread that's assigned to the request. 
```suggestion
    If no value is found, or one is found but is invalid, we generate and store a new GUID on the thread.
```
or something like that?